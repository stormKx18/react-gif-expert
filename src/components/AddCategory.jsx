import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        const cleanInput = inputValue.trim()
        if(cleanInput.length<=1) return;
        //setCategories((categories)=>[inputValue,...categories]);
        onNewCategory(cleanInput)
        setInputValue('')
    }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input 
        type="text" 
        placeholder="Search gifs..."
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

