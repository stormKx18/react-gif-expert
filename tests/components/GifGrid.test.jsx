import { render, screen } from '@testing-library/react'
import { GifGrid } from '../../src/components/GifGrid'
import { useFetchGifs } from '../../src/hooks/useFetchGifs'

jest.mock('../../src/hooks/useFetchGifs')

describe('Test for <GifGrid />', () => {
    const category = 'spiderman'

    test('Component should match snapshot ', () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        })
        const {container} = render(<GifGrid category={category}/>);
        expect(container).toMatchSnapshot();
    })

    test('Component should show loading message at start', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });
        render(<GifGrid category={category}/>)
        expect(screen.getByText('Loading ...')).toBeTruthy();
        expect(screen.getByText(category)).toBeTruthy();
    })

    test('Component should show items when they are loaded from useFetchGifs', () => {
        
        const gifs =[
            {
                id:'ABC',
                title: 'Wolverine',
                url: 'https://localhost/wolverine.jpg'
            },
                        {
                id:'DEF',
                title: 'Spongebob',
                url: 'https://localhost/spongebob.jpg'
            },
                        {
                id:'GHI',
                title: 'Arnold',
                url: 'https://localhost/Arnold.jpg'
            },
        ]

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        });

        render(<GifGrid category={category}/>)
        // screen.debug()
        expect(screen.getAllByRole('img').length).toBe(gifs.length)

    })
    
    
  
})
