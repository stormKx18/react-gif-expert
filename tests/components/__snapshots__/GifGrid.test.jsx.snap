// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifGrid /> Component should match snapshot  1`] = `
<div>
  <h3>
    spiderman
  </h3>
  <h2>
    Loading ...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
